File: SpotifyAngular/src/app/Models/songModel.ts
import { ArtistModel } from "./artistModel";

export type SongModel={
    id:number,
    name:string,
    uri:string,
    duration:number,
    image?:string,
    artists:ArtistModel[],
    selectedArtists?:number[],
    addedOn?:Date
}

export type SongDataSource={
    position:number,
    id:number,
    name:string,
    uri:string,
    duration:number,
    image:string|null,
    artists:string|null,
    songArtists:ArtistModel[]|null
}

File: SpotifyAPI.DAL/Models/SongModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SpotifyAPI.DAL.Models
{
    public class SongModel
    {
        public int? Id { get; set; }
        public string? Name { get; set; }
        public string? Uri { get; set; }
        public int? Duration { get; set; }
        public int? PlayListId { get; set; }
        public string? Image { get; set; }
        public PlaylistModel? Playlist { get; set; }
        public DateTime? AddedOn { get; set; }
        public List<ArtistModel>? Artists { get; set; }

    }
    public class SongException : Exception
    {
        public SongException(string message) : base(message) { }
    }
}
